core/src/main/java/ai/timefold/solver/core/api/solver/SolutionManager.java

```diff
@@ -137,10 +137,122 @@ default ScoreAnalysis<Score_> analyze(Solution_ solution, ScoreAnalysisFetchPoli
+     * In this case, it will fail either directly by throwing an exception,
+     * or indirectly by not providing correct data.
+     *
+     * <p>
+     * When an element is tested,
+     * a score is calculated over the entire solution with the element in place,
+     * also called a placement.
+     * The proposition function is also called at that time,
+     * allowing the user to extract any information from the current placement;
+     * the extracted information is called the proposition.
+     * After the proposition is extracted,
+     * the solution is returned to its original state,
+     * resetting all changes made by the fitting.
+     * This has a major consequence for the proposition, if it is a planning entity:
+     * planning entities contain live data in their planning variables,
+     * and that data will be erased when the next placement is tested for fit.
+     * In this case,
+     * the proposition function needs to make defensive copies of everything it wants to return,
+     * such as values of shadow variables etc.
+     *
+     * <p>
+     * Example: Consider a planning entity Shift, with a variable "employee".
+     * Let's assume we have two employees to test for fit, Ann and Bob,
+     * and a single Shift instance to fit them into, {@code mondayShift}.
+     * The proposition function will be called twice,
+     * once as {@code mondayShift@Ann} and once as {@code mondayShift@Bob}.
+     * Let's assume the proposition function returns the Shift instance in its entirety.
+     * This is what will happen:
+     *
+     * <ol>
+     * <li>Calling propositionFunction on {@code mondayShift@Ann} results in proposition P1: {@code mondayShift@Ann}</li>
+     * <li>Placement is cleared, {@code mondayShift@Bob} is now tested for fit.</li>
+     * <li>Calling propositionFunction on {@code mondayShift@Bob} results in proposition P2: {@code mondayShift@Bob}</li>
+     * <li>Proposition P1 and P2 are now both the same {@code mondayShift},
+     * which means Bob is now assigned to both of them.
+     * This is because both propositions operate on the same entity,
+     * and therefore share the same state.
+     * </li>
+     * <li>The placement is then cleared again,
+     * both elements have been tested,
+     * and solution is returned to its original order.
+     * The propositions are then returned to the user,
+     * who notices that both P1 and P2 are {@code mondayShift@null}.
+     * This is because they shared state,
+     * and the original state of the solution was for Shift to be unassigned.
+     * </li>
+     * </ol>
+     *
+     * If instead the proposition function returned Ann and Bob directly, the immutable planning variables,
+     * this problem would have been avoided.
+     * Alternatively, the proposition function could have returned a defensive copy of the Shift.
+     *
+     * @param solution never null; must be fully initialized except for one entity or element
```

Comment:
This param explanation is unclear to me. Can we explicitly state the difference between an initialized and uninitialized solution?